sql/core/src/main/scala/org/apache/spark/sql/GroupedDataset.scala

```diff
@@ -137,8 +137,19 @@ class GroupedDataset[K, T] private[sql](
   }
 
   // To ensure valid overloading.
-  protected def agg(expr: Column, exprs: Column*): DataFrame =
-    groupedData.agg(expr, exprs: _*)
+  @scala.annotation.varargs
+  def agg(expr: Column, exprs: Column*): DataFrame =
```

Comment:
@rxin I think we need to expose this for the java users to work, while still getting nice inference for scala